### PR TITLE
Create potential_energy.py

### PR DESCRIPTION
**Pull request by SparshRastogi**
Finding the gravitational potential energy of an object with reference to the earth, by taking its mass and height above the ground as input

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: physics/potential_energy.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from doctest import testmod
+
+import scipy.constants as cons
```
Comment (ChrisO345) [inline physics/potential_energy.py:3]:
```suggestion
from scipy.constants import g
```

File: physics/potential_energy.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    >>> potential_energy(20,100)
+    19613.3
+    """
+    if mass < 0:
+        raise ValueError("The mass of a body cannot be negative")
+        # handling of negative values of mass
+    if height < 0:
+        raise ValueError("The height above the ground cannot be negative")
+        # handling of negative values of height
+    return mass * cons.g * height
```
[the remark this answers is not in the document]
Comment (SparshRastogi) [reply physics/potential_energy.py:58]: @ChrisO345 thank you for the suggestion

File: physics/potential_energy.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+from doctest import testmod
+
```
Comment (CaedenPH) [inline physics/potential_energy.py:2]:
```suggestion
```
Moving this within `if __name__ == "__main__"`...

File: physics/potential_energy.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    if mass < 0:
+        raise ValueError("The mass of a body cannot be negative")
+        # handling of negative values of mass
+    if height < 0:
+        raise ValueError("The height above the ground cannot be negative")
+        # handling of negative values of height
+    return mass * g * height
+
+
+if __name__ == "__main__":
```
Comment (CaedenPH) [inline physics/potential_energy.py:61]:
```suggestion
if __name__ == "__main__":
    from doctest import testmod
    
```

File: physics/potential_energy.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+U=-GMm/R
+where M and m are the masses of the two particles, R is the distance
+between them, and G is the gravitational constant.
+Close to the Earth's surface, the gravitational field is approximately
+constant, and the gravitational potential energy of an object reduces to
+U=mgh
+where m is the object's mass, g=GM/R² is the gravity of Earth, and h is
+the height of the object's center of mass above a chosen reference level.
+
+
```
Comment (CaedenPH) [inline physics/potential_energy.py:29]:
```suggestion
```

File: physics/potential_energy.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    >>> potential_energy(0,0)
+    0.0
+    >>> potential_energy(2,8)
+    156.9064
+    >>> potential_energy(20,100)
+    19613.3
+    """
+    if mass < 0:
+        raise ValueError("The mass of a body cannot be negative")
+        # handling of negative values of mass
```
Comment (CaedenPH) [inline physics/potential_energy.py:54]:
```suggestion
        # handling of negative values of mass
        raise ValueError("The mass of a body cannot be negative")
```
it's more confusing if you put it after the `raise`

File: physics/potential_energy.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    if height < 0:
+        raise ValueError("The height above the ground cannot be negative")
+        # handling of negative values of height
```
Comment (CaedenPH) [inline physics/potential_energy.py:57]:
```suggestion
        # handling of negative values of height
        raise ValueError("The height above the ground cannot be negative")
```
Same here

File: physics/potential_energy.py (hunk 6 above)
Comment (SparshRastogi) [reply physics/potential_energy.py:51]: Yeah,I like your way

**Review by CaedenPH — APPROVED** [review]
@ChrisO345 I believe he made the changes you asked for, you could review again please?

**Review by ChrisO345 — DISMISSED** [review]
Could you please add scipy to the requirements.txt as part of this PR

**Comment by SparshRastogi** [comment]
> Could you please add scipy to the requirements.txt as part of this PR

@ChrisO345 for that I would need to create a directory for this file but all the other files are just individual Python files in this directory,and in the pull request also I acknowledged that I am creating a single file inside an existing directory,so I don't think it would be great

**Review by ChrisO345 — APPROVED** [review]

**Pull request merged**